pkg/diagnostics/log/log.go

```diff
@@ -83,112 +136,138 @@ type Msg map[string]interface{}
  */
 
 var (
-	ErrorLevel  = Level{0, "error", "ERROR: ", ct.Red, true}   // Something is definitely wrong
-	WarnLevel   = Level{1, "warn", "WARN:  ", ct.Yellow, true} // Likely to be an issue but maybe not
+	ErrorLevel  = Level{4, "error", "ERROR: ", ct.Red, true}   // Something is definitely wrong
+	WarnLevel   = Level{3, "warn", "WARN:  ", ct.Yellow, true} // Likely to be an issue but maybe not
 	NoticeLevel = Level{2, "note", "[Note] ", ct.White, false} // Introductory / summary
-	InfoLevel   = Level{3, "info", "Info:  ", ct.None, false}  // Just informational
-	DebugLevel  = Level{4, "debug", "debug: ", ct.None, false} // Extra verbose
+	InfoLevel   = Level{1, "info", "Info:  ", ct.None, false}  // Just informational
+	DebugLevel  = Level{0, "debug", "debug: ", ct.None, false} // Extra verbose
 )
 
 // Provide a summary at the end
 func (l *Logger) Summary() {
 	l.Notice("summary", "\nSummary of diagnostics execution:\n")
 	if l.warningsSeen > 0 {
-		l.Noticem("sumWarn", Msg{"tmpl": "Warnings seen: {{.num}}", "num": l.warningsSeen})
+		l.Noticef("sumWarn", "Warnings seen: %d", l.warningsSeen)
 	}
 	if l.errorsSeen > 0 {
-		l.Noticem("sumErr", Msg{"tmpl": "Errors seen: {{.num}}", "num": l.errorsSeen})
+		l.Noticef("sumErr", "Errors seen: %d", l.errorsSeen)
 	}
 	if l.warningsSeen == 0 && l.errorsSeen == 0 {
 		l.Notice("sumNone", "Completed with no errors or warnings seen.")
 	}
 }
 
-func (l *Logger) Log(level Level, id string, msg Msg) {
-	if level.Level > l.level.Level {
+func (l *Logger) LogMessage(level Level, message Message) {
+	// if there's no logger, return silently
+	if l == nil {
 		return
 	}
-	msg["id"] = id // TODO: use to retrieve template from elsewhere
-	// if given a template, convert it to text
-	if tmpl, exists := msg["tmpl"]; exists {
-		var buff bytes.Buffer
-		if tmplString, assertion := tmpl.(string); !assertion {
-			msg["templateErr"] = fmt.Sprintf("Invalid template type: %T", tmpl)
-			msg["templateId"] = id
-			msg["id"] = "tmplErr"
-		} else {
-			parsedTmpl, err := template.New(id).Parse(tmplString)
-			if err != nil {
-				msg["templateErr"] = err.Error()
-				msg["templateId"] = id
-				msg["id"] = "tmplErr"
-			} else if err = parsedTmpl.Execute(&buff, msg); err != nil {
-				msg["templateErr"] = err.Error()
-				msg["templateId"] = id
-				msg["id"] = "tmplErr"
-			} else {
-				msg["text"] = buff.String()
-				delete(msg, "tmpl")
-			}
-		}
-	}
+
+	// track how many of every type we've seen (probably unnecessary)
```

Comment:
I don't consider it unnecessary... when you have a wall of text go by, a summary at the end makes it a lot more consumable.